src/uwp/Microsoft.Build.Net.CoreRuntimeTask/ResourceHandlingTask.cs

```diff
@@ -267,18 +323,82 @@ private void WriteStateFile(string stateFile, ResourceHandlingState state)
             }
         }
 
-        private string ExtractFrameworkAssemblyResW(string assemblyFilePath, out bool containsResources)
+        private ResWInfo ExtractResourcesFromStream(Stream stream, IAssembly assembly, string resourceFileName, bool containsFrameworkResources)
         {
-            string assemblyName;
-            using (Stream stream = ExtractFromAssembly(assemblyFilePath, out assemblyName, out containsResources))
+            string reswFilePath;
+            string resourceIndexName;
+            string neutralResourceLanguage = "";
+
+            if (containsFrameworkResources)
             {
-                if (stream == null)
-                    return null;
+                reswFilePath = OutResWPath + Path.AltDirectorySeparatorChar + resourceFileName + ".resw";
+                resourceIndexName = resourceFileName;
+                neutralResourceLanguage = "en-US";
+            }
+            else {
+                string culturePath = "";
+                string culture = assembly.Culture;
+                string assemblyName = assembly.Name.Value;
+
+                if (!String.IsNullOrEmpty(culture))
+                {
+                    culturePath = culture + Path.DirectorySeparatorChar;
+                }
+                else if (TryGetNeutralResourcesLanguageAttribute(assembly, out neutralResourceLanguage))
+                {
+                    culturePath = neutralResourceLanguage + Path.DirectorySeparatorChar;
+                }
+                else
```

Comment:
> else [](start = 15, length = 5)

please comment this else

Reply:
Actually should we never hit this case here? I mean maybe we can keep the else and assert inside it?

---
In reply to: [135931755](https://github.com/dotnet/core-setup/pull/3114#discussion_r135931755) [](ancestors = 135931755)

Reply:
Adding an assert.